Quotient/threadview.cpp

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: 2024 James Graham <james.h.graham@protonmail.com>
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "threadview.h"
+
+#include "events/roommessageevent.h"
+
+using namespace Quotient;
+
+void ThreadView::add(std::unique_ptr<Thread> thread)
+{
+    _threads.push_back(std::move(thread));
+}
+
+bool ThreadView::erase(const QString& threadRootId)
+{
+    auto threadIt = std::find_if(_threads.begin(), _threads.end(), [threadRootId](const auto& thread) { return thread->threadRootId() == threadRootId; });
+    if (threadIt == _threads.end()) {
+        return false;
+    }
+     _threads.erase(threadIt);
+    return true;
```

Comment:
```suggestion
    return std::erase_if(_threads, [threadRootId](const auto& thread) { return thread->threadRootId() == threadRootId; }) > 0;
```
Actually, I suggest making a small helper `checkThreadRoot()` function that would return this lambda, so that you could do `return std::erase_if(_threads, checkThreadRoot(threadRootId)) > 0;`, and so on across all the methods below. The function could look like:
```cpp
namespace {
inline auto checkThreadRoot(const QString& threadRootId) {
    return [threadRootId](const std::unique_ptr<Thread>& thread) { return threadRootId == thread->threadRootId(); };
}
}
```